cupy/_statistics/correlation.py

```diff
@@ -131,13 +145,62 @@ def cov(a, y=None, rowvar=True, bias=False, ddof=None):
     if ddof is None:
         ddof = 0 if bias else 1
 
-    fact = X.shape[1] - ddof
+    w = None
+    if fweights is not None:
+        fweights = cupy.asarray(fweights, dtype=float)
+        if not cupy.all(fweights == cupy.around(fweights)):
+            raise TypeError(
+                "fweights must be integer")
+        if fweights.ndim > 1:
+            raise RuntimeError(
+                "cannot handle multidimensional fweights")
+        if fweights.shape[0] != X.shape[1]:
+            raise RuntimeError(
+                "incompatible numbers of samples and fweights")
+        if any(fweights < 0):
```

Comment:
This also causes device synchronization. 
You apply an operation `<` to all the elements of the array in the GPU, and then move the result to the CPU to execute the python `any` to drive that `if`.

To avoid this we are flexible and allow negative results although it can produce errors, just a note in the docstring saying that the function does not error with negative values for performance should be enough.

Reply:
This would also apply to `if any(aweights < 0)` then.